sale_order_general_discount_payment_term/models/account_payment_term.py

```diff
@@ -0,0 +1,12 @@
+from odoo import fields, models
+
+
+class AccountPaymentTerm(models.Model):
+    _inherit = "account.payment.term"
+
+    sale_discount = fields.Float(
+        digits="Discount",
+        string="Discount (%)",
+        company_dependent=True,
+        track_visibility="onchange",
```

Comment:
as you correctly spotted in the other comment, I was looking at v12 code

`track_visibility` was a v12 option, in v14 it's simply `tracking=True`